src/projection_client.rs

```diff
@@ -0,0 +1,629 @@
+
+        crate::commands::configure_auth_req(&mut req, credentials.cloned());
+
+        self.client
+            .execute(|handle| async move {
+                let mut client =
+                    projections::projections_client::ProjectionsClient::new(handle.channel.clone());
+
+                let _ = client.enable(req).await?;
+
+                Ok(())
+            })
+            .await
+    }
+
+    pub async fn reset_projection<Name>(
+        &self,
+        name: Name,
+        write_checkpoint: bool,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        let options = projections::reset_req::Options {
+            name: name.as_ref().to_string(),
+            write_checkpoint,
+        };
+
+        let req = projections::ResetReq {
+            options: Some(options),
+        };
+
+        let mut req = tonic::Request::new(req);
+
+        crate::commands::configure_auth_req(&mut req, credentials.cloned());
+
+        self.client
+            .execute(|handle| async move {
+                let mut client =
+                    projections::projections_client::ProjectionsClient::new(handle.channel.clone());
+
+                let _ = client.reset(req).await?;
+
+                Ok(())
+            })
+            .await
+    }
+
+    pub async fn disable_projection<Name>(
+        &self,
+        name: Name,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        self.disable_projection_internal(name, true, credentials)
+            .await
+    }
+
+    pub async fn abort_projection<Name>(
+        &self,
+        name: Name,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        self.disable_projection_internal(name, false, credentials)
+            .await
+    }
+
+    async fn disable_projection_internal<Name>(
+        &self,
+        name: Name,
+        write_checkpoint: bool,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        let options = projections::disable_req::Options {
+            name: name.as_ref().to_string(),
+            write_checkpoint,
+        };
+
+        let req = projections::DisableReq {
+            options: Some(options),
+        };
+
+        let mut req = tonic::Request::new(req);
+
+        crate::commands::configure_auth_req(&mut req, credentials.cloned());
+
+        self.client
+            .execute(|handle| async move {
+                let mut client =
+                    projections::projections_client::ProjectionsClient::new(handle.channel.clone());
+
+                let _ = client.disable(req).await?;
+
+                Ok(())
+            })
+            .await
+    }
+
+    pub async fn projection_state<Name, Partition, A>(
+        &self,
+        name: Name,
+        partition: Partition,
```

Comment:
**CONCERN:** In .NET, this is optional and set to an empty string.

Reply:
There is no default parameter in Rust and there are also several string representations too (`String` and `&str`). That type-signature allows us to use both (Rust doesn't support function overloading)